source/adapters/level_zero/queue.cpp

```diff
@@ -1403,6 +1405,18 @@ ur_queue_handle_t_::executeCommandList(ur_command_list_ptr_t CommandList,
       // If we don't have host visible proxy then signal event if needed.
       this->signalEventFromCmdListIfLastEventDiscarded(CommandList);
     }
+    // Append Signalling of the inner events at the end of the batch
+    for (auto &Event : CommandList->second.EventList) {
+      if (Event->IsInnerBatchedEvent) {
+        if (AppendBarrierNeeded) {
+          ZE2UR_CALL(zeCommandListAppendBarrier,
```

Comment:
can you add a comment why a barrier is needed? this can only be true if this is an out-of-order queue, right?

Reply:
correct, I can add a comment. This barrier is only needed when there are these events which are only there if this is out of order.